src/main/java/uk/gov/pay/directdebit/webhook/gocardless/api/WebhookParser.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.pay.directdebit.webhook.gocardless.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import uk.gov.pay.directdebit.app.logger.PayLoggerFactory;
+import uk.gov.pay.directdebit.payments.model.GoCardlessEvent;
+import uk.gov.pay.directdebit.webhook.gocardless.exception.WebhookParserException;
+
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class WebhookParser {
+    private static final Logger LOGGER = PayLoggerFactory.getLogger(WebhookParser.class);
+
+    private ObjectMapper objectMapper;
+
+    @Inject
+    public WebhookParser(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public enum HandledResourceType {
+        PAYMENTS, MANDATES;
+
+        public static HandledResourceType fromString(String type) {
+            for (HandledResourceType typeEnum : HandledResourceType.values()) {
+                if (typeEnum.toString().equalsIgnoreCase(type)) {
+                    return typeEnum;
+                }
+            }
+            LOGGER.warn("Received webhook from gocardless with unhandled resource_type: {}", type);
```

Comment:
at least in the beginning we might receive unhandled webhooks because we haven't played the relevant stories / completely understood the flow, that's why I thought of keeping it as a warning and then switch to info once we reach a steady state :)